src/main/java/com/redhat/devtools/lsp4ij/console/actions/ApplyLanguageServerSettingsAction.java

```diff
@@ -41,11 +51,55 @@ public void actionPerformed(@NotNull AnActionEvent e) {
 
     @Override
     public void update(@NotNull AnActionEvent e) {
-        e.getPresentation().setEnabled(languageServerView.isModified());
+        boolean modified = languageServerView.isModified();
+        boolean isShown = languageServerView.isSaveTipShown();
+
+        // Only show the tooltip once per configuration change
+        if (modified && !isShown) {
+            showBalloon();
+        }
+        // If configuration is returned to match the unmodified state, reset the tooltip to be shown again
+        if (!modified) {
+            languageServerView.isSaveTipShown(false);
+        }
+
+        e.getPresentation().setEnabled(modified);
     }
 
     @Override
     public @NotNull ActionUpdateThread getActionUpdateThread() {
         return ActionUpdateThread.BGT;
     }
+
+    private void showBalloon() {
+        languageServerView.isSaveTipShown(true);
+        JBLabel jbPanel = new JBLabel(LanguageServerBundle.message("action.lsp.detail.apply.balloon"));
+
+        BalloonBuilder builder = JBPopupFactory.getInstance()
+                .createBalloonBuilder(jbPanel)
+                .setFadeoutTime(800)
+                .setHideOnAction(false);
+
+        Balloon balloon = builder.createBalloon();
+
+        ActionToolbarImpl actionToolbarComponent = null;
+        var components = languageServerView.getComponent().getComponents();
+        for (Component component : components) {
+            if (ACTION_TOOLBAR_COMPONENT_NAME.equals(component.getName())) {
+                actionToolbarComponent = (ActionToolbarImpl) component;
+            }
+        }
+
+        if (actionToolbarComponent == null) {
+            return;
+        }
+        Component applyComponent = actionToolbarComponent.getComponent(0);
+
+        // Move the position by 1/2 of the component width, because the icon is not centered
+        Point point = new Point(applyComponent.getWidth()/2 ,0);
+        RelativePoint displayPoint = new RelativePoint(applyComponent, point);
+
+        // Use invokeLater to prevent the balloon from resizing when it is shown
+        SwingUtilities.invokeLater(() -> balloon.show(displayPoint, Balloon.Position.above));

```

Comment:
I think it is better to use `ApplicationManager.getAppllication().invokeLater` right?

Reply:
Yep, will fix